pkg/networkservice/connectioncontext/dnscontext/client.go

```diff
@@ -96,22 +96,51 @@ func (c *dnsContextClient) Close(ctx context.Context, conn *networkservice.Conne
 }
 
 func (c *dnsContextClient) restoreResolvConf() {
-	originalResolvConf, err := os.ReadFile(c.storedResolvConfigPath)
-	if err != nil || len(originalResolvConf) == 0 {
+	bytes, err := os.ReadFile(c.resolveConfigPath)
+	if err != nil {
 		return
 	}
-	_ = os.WriteFile(c.resolveConfigPath, originalResolvConf, os.ModePerm)
+
+	commentLines := make([]string, 0)
+	for _, line := range strings.Split(string(bytes), "\n") {
+		if strings.HasPrefix(line, "#") {
+			commentLines = append(commentLines, line[1:])
+		}
+	}
+
+	commentedPart := strings.Join(commentLines, "\n")
+	parsedConfig, _ := dns.ClientConfigFromReader(strings.NewReader(commentedPart))
+
+	if len(parsedConfig.Servers) == 0 {
+		return
+	}
+
+	_ = os.WriteFile(c.resolveConfigPath, []byte(commentedPart), os.ModePerm)
 }
 
 func (c *dnsContextClient) storeOriginalResolvConf() {
-	if _, err := os.Stat(c.storedResolvConfigPath); err == nil {
+	bytes, err := os.ReadFile(c.resolveConfigPath)
+	originalResolvConfig := string(bytes)
```

Comment:
It will be better to check for error before `originalResolvConfig := string(bytes)`